src/server/service/kv.rs

```diff
@@ -811,6 +811,47 @@ impl<T: RaftStoreRouter + 'static, E: Engine> tikvpb_grpc::Tikv for Service<T, E
         ctx.spawn(future);
     }
 
+    fn unsafe_destroy_range(
+        &self,
+        ctx: RpcContext,
+        mut req: UnsafeDestroyRangeRequest,
+        sink: UnarySink<UnsafeDestroyRangeResponse>,
+    ) {
+        let timer = GRPC_MSG_HISTOGRAM_VEC
+            .unsafe_destroy_range
+            .start_coarse_timer();
+
+        // DestroyRange is a very dangerous operation. We don't allow passing MIN_KEY as start, or
+        // MAX_KEY as end here.
+        assert!(!req.get_start_key().is_empty());
+        assert!(!req.get_end_key().is_empty());
+
+        let (cb, f) = paired_future_callback();
+        let res = self.storage.async_unsafe_destroy_range(
+            req.take_context(),
+            Key::from_raw(&req.take_start_key()),
+            Key::from_raw(&req.take_end_key()),
+            cb,
+        );
+
+        let future = AndThenWith::new(res, f.map_err(Error::from))
+            .and_then(|v| {
+                let mut resp = UnsafeDestroyRangeResponse::new();
+                // Region error is impossible here.
+                if let Err(e) = v {
+                    resp.set_error(format!("{}", e));
+                }
+                sink.success(resp).map_err(Error::from)
+            })
+            .map(|_| timer.observe_duration())
```

Comment:
`map` or `then`?

Reply:
It just keeps the same as other functions in this file.